effect/RpcCall.ts

```diff
@@ -0,0 +1,47 @@
+import { Config } from "../config/mod.ts";
+import * as Z from "../deps/zones.ts";
+import * as rpc from "../rpc/mod.ts";
+import { RpcError } from "./common.ts";
+import { rpcClient } from "./core/rpcClient.ts";
+
+export class RpcCall<
+  Methods extends rpc.ProviderMethods,
+  MethodName extends Z.$<Extract<keyof Methods, string>>,
+  Params extends Z.Ls$<Parameters<Methods[Z.T<MethodName>]>>,
+> extends Z.Name {
+  root;
+
+  constructor(
+    config: Config<string, Methods>,
+    methodName: MethodName,
+    params: [...Params],
+  ) {
+    super();
+    const client = rpcClient(config);
+    const deps = Z.ls(client, methodName, ...params);
+    this.root = Z.call(
+      Z.ls(deps, Z.rc(client, deps)),
+      async ([[client, methodName, ...params], rc]) => {
+        const result = await client.call(
+          methodName,
+          params as Parameters<(Methods & rpc.ProviderMethods)[Z.T<MethodName>]>,
+        );
+        if (result.error) {
+          return new RpcError({
+            ...result.error,
+            attempt: {
+              methodName,
+              params,
+            },
+          });
+        }
+        if (rc() == 1) {
+          const close = await client.close();
+          if (close instanceof Error) return close;
+        }
```

Comment:
I added a couple of `console.log` to `RpcCall`, `rpcClient` and `example/watch_blocks` and the output was

```
root@480c9297b2b1:/workspaces/capi# deno task run test_ctx.ts deno task run examples/watch_blocks.ts
Warning deno task is unstable and may drastically change in the future
Task run deno run -A --no-check=remote "test_ctx.ts" "deno" "task" "run" "examples/watch_blocks.ts"
Check file:///workspaces/capi/test_ctx.ts
Warning deno task is unstable and may drastically change in the future
Task run deno run -A --no-check=remote "examples/watch_blocks.ts"
Check file:///workspaces/capi/examples/watch_blocks.ts
rpc.stdClient
ws.onOpen
result
rpc.stdClient
ws.onOpen
result
rpc.stdClient
result
rpc.stdClient
ws.onOpen
closing..
ws.onOpen
close instanceof Error false
// block data and count ==  0
closing..
close instanceof Error false
// block data and count ==  1
closing..
close instanceof Error false
// block data and count ==  2
rpc.stdClient
ws.onOpen
closing..
close instanceof Error false
```

It seems that
- `rpc.stdClient(config)` is invoked many times
- as a result `rc() == 1` is always true, that's why `client.close()` is invoked more than once


This would be the patch for above `console.log` entries
```
diff --git a/effect/RpcCall.ts b/effect/RpcCall.ts
index 8059e3a..465c8f5 100644
--- a/effect/RpcCall.ts
+++ b/effect/RpcCall.ts
@@ -36,7 +36,10 @@ export class RpcCall<
           });
         }
         if (rc() == 1) {
+          console.log("closing..");
           const close = await client.close();
+          console.log("close instanceof Error", close instanceof Error);
+          // console.log("result", result);
           if (close instanceof Error) return close;
         }
         // TODO: should this effect implicitly index into `result`?
diff --git a/effect/core/rpcClient.ts b/effect/core/rpcClient.ts
index 742e9f1..8a40f51 100644
--- a/effect/core/rpcClient.ts
+++ b/effect/core/rpcClient.ts
@@ -7,5 +7,6 @@ export function rpcClient<C extends Config>(config: C) {
 }
:...skipping...
diff --git a/effect/RpcCall.ts b/effect/RpcCall.ts
index 8059e3a..465c8f5 100644
--- a/effect/RpcCall.ts
+++ b/effect/RpcCall.ts
@@ -36,7 +36,10 @@ export class RpcCall<
           });
         }
         if (rc() == 1) {
+          console.log("closing..");
           const close = await client.close();
+          console.log("close instanceof Error", close instanceof Error);
+          // console.log("result", result);
           if (close instanceof Error) return close;
         }
         // TODO: should this effect implicitly index into `result`?
diff --git a/effect/core/rpcClient.ts b/effect/core/rpcClient.ts
index 742e9f1..8a40f51 100644
--- a/effect/core/rpcClient.ts
+++ b/effect/core/rpcClient.ts
@@ -7,5 +7,6 @@ export function rpcClient<C extends Config>(config: C) {
 }
 
 function impl<C extends Config>(config: C) {
+  console.log("rpc.stdClient");
   return rpc.stdClient(config);
 }
diff --git a/examples/watch_blocks.ts b/examples/watch_blocks.ts
index 29ae6d1..ec8420c 100644
--- a/examples/watch_blocks.ts
```